pkg/storage/compliance/tests.go

```diff
@@ -120,10 +130,57 @@ func testDelete(t *testing.T, b storage.Backend) {
 	require.Equal(t, false, exists)
 }
 
+func testCatalog(t *testing.T, b storage.Backend) {
+	ctx := context.Background()
+
+	mock := getMockModule()
+	zipBts, _ := ioutil.ReadAll(mock.Zip)
+	modname := "github.com/gomods/testCatalogModule"
+	for i := 0; i < 1005; i++ {
+		ver := fmt.Sprintf("v1.2.%04d", i)
+		b.Save(ctx, modname, ver, mock.Mod, bytes.NewReader(zipBts), mock.Info)
+	}
+	defer func() {
+		for i := 0; i < 1005; i++ {
+			ver := fmt.Sprintf("v1.2.%04d", i)
+			b.Delete(ctx, modname, ver)
+		}
+	}()
+
+	allres, next, err := b.Catalog(ctx, "", 1001)
+
+	require.NoError(t, err)
+	require.Equal(t, 1001, len(allres))
+
+	res, next, err := b.Catalog(ctx, next, 50)
+	allres = append(allres, res...)
+	require.NoError(t, err)
+	require.Equal(t, 4, len(res))
+	require.Equal(t, "", next)
+
+	sort.Slice(allres, func(i, j int) bool {
+		if allres[i].Module == allres[j].Module {
```

Comment:
From the tests it looks like we will never hit the condition. Should we just check for uniqueness? Please cporrect me if I am wrong.

Reply:
I think checking for duplicates would be a nice to have